server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -427,6 +427,7 @@ export function configureWebSocketServices(
 			const isSummarizer = messageClient.details?.type === summarizerClientType;
 			messageClient.user = claims.user;
 			messageClient.scopes = claims.scopes;
+			messageClient.mode = hasWriteAccess(claims.scopes) ? "write" : "read";
```

Comment:
```suggestion
			messageClient.mode = hasWriteAccess(claims.scopes) ? client.mode : "read";
```

Reply:
i think a client with write access should still be able to join as a read client

Reply:
there is also a similar function, isWriter, that could maybe be used here

Reply:
Good point, fixed to use isWriter